libs/net/vmspec.py

```
@@ -299,3 +300,33 @@ def _vmi_condition_not_set(existing_conditions: list[ResourceField], required_co
+
+    Patches the VM spec atomically, then waits for the MigrationRequired condition to
+    appear (change detected) and disappear (migration completed).
+
+    Args:
+        vm: The virtual machine to update.
+        nad_name_by_net: Mapping of interface name to new NAD name.
+    """
+    resource_version = vm.vmi.instance.metadata.resourceVersion
+    networks = vm.template_spec.networks
+    if not networks:
+        raise ValueError(f"VM {vm.name} has no template networks to update")
```

Ruff (0.15.15)

[warning] Avoid specifying long messages outside the exception class

(TRY003)